contracts/oracle/src/contract/feed.rs

```diff
@@ -127,20 +193,148 @@ pub fn try_query_alarms<OracleBase>(
+                                "children":[
+                                    {{"value":[5,"{juno}"]}},
+                                    {{"value":[6,"{cro}"]}}
+                                ]
+                            }}
+                        ]
+                    }}
+                ]
+            }}"#
+        ))
+        .unwrap()
+    }
+
+    #[test]
+    fn all_prices_iter() {
+        let mut storage = MockStorage::new();
+        let env = testing::mock_env();
+        let tree = test_case();
+        let tree = SupportedPairs::<TheCurrency>::new(tree.into_tree()).unwrap();
+        tree.save(&mut storage).unwrap();
+
+        let config = Config::new(
+            Percent::HUNDRED,
+            Duration::from_secs(5),
+            10,
+            Percent::from_percent(50),
+        );
+
+        let oracle: Feeds<TheCurrency> = Feeds::with(config);
+
+        oracle
+            .feed_prices(
+                &mut storage,
+                env.block.time,
+                &Addr::unchecked("feeder"),
+                &[
+                    price::total_of(Coin::<Wbtc>::new(1))
+                        .is(Coin::<TheCurrency>::new(1))
+                        .into(),
+                    price::total_of(Coin::<Atom>::new(2))
+                        .is(Coin::<TheCurrency>::new(1))
+                        .into(),
+                    price::total_of(Coin::<Weth>::new(1))
+                        .is(Coin::<Wbtc>::new(1))
+                        .into(),
+                    price::total_of(Coin::<Osmo>::new(1))
+                        .is(Coin::<Atom>::new(1))
+                        .into(),
+                    price::total_of(Coin::<Cro>::new(3))
+                        .is(Coin::<Osmo>::new(1))
+                        .into(),
+                    price::total_of(Coin::<Juno>::new(1))
+                        .is(Coin::<Osmo>::new(1))
+                        .into(),
+                ],
+            )
+            .unwrap();
+
+        let prices: Vec<_> = oracle
+            .all_prices_iter(&storage, &tree, env.block.time, 1)
+            .unwrap()
+            .collect();
+
+        let expected: Vec<SpotPrice> = vec![
+            price::total_of(Coin::<Wbtc>::new(1))
+                .is(Coin::<TheCurrency>::new(1))
+                .into(),
+            price::total_of(Coin::<Weth>::new(1))
+                .is(Coin::<TheCurrency>::new(1))
+                .into(),
+            price::total_of(Coin::<Atom>::new(2))
+                .is(Coin::<TheCurrency>::new(1))
+                .into(),
+            price::total_of(Coin::<Osmo>::new(2))
+                .is(Coin::<TheCurrency>::new(1))
+                .into(),
+            price::total_of(Coin::<Juno>::new(2))
+                .is(Coin::<TheCurrency>::new(1))
+                .into(),
+            price::total_of(Coin::<Cro>::new(6))
+                .is(Coin::<TheCurrency>::new(1))
+                .into(),
+        ];
+
```

Comment:
pls remove the code duplications and make it more readable by defining a local fn